csrc/codegen.cpp

```diff
@@ -2995,9 +2987,17 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
           break;
         case MemoryType::Local: {
           auto va = kernel_->summary().vectorized_accesses;
+          unsigned int vect_factor = 0;
           if (va.find(tv) != va.end()) {
+            vect_factor = va.at(tv);
+          } else if (
+              kernel_->summary().num_grouped_iterations > 1 &&
+              ir_utils::isConsumedByIterGroupedReduction(tv)) {
+            vect_factor = kernel_->summary().num_grouped_iterations;
```

Comment:
In general, `Kernel` should already have the correct vectorization factor and `CudaCodeGen` should just be a straightforward printer. Is there any specific reason for this case?

Reply:
`Kernel` only has vectorized factor for global memory access, not for shared memory access.I added comments for clarity.
```
Should use aligned array of registers when:
(1) vectorized ld/st with global memory, tv exists in kernel summary vectorized_accesses.
(2) vectorized ld/st with shared memory, tv is input to iteration grouped reduction and vectorized in runtime function.
```
Another option, is adding these tvs with vectorized shared memory access to `kernel_->summary().vectorized_accesses`.

Reply:
It seems strange to me that the kernel summary only has info about global memory tensors but not shared memory. Is there any reason for that?

Reply:
> It seems strange to me that the kernel summary only has info about global memory tensors but not shared memory. Is there any reason for that?

To be more precise, should be `kernel summary doesn't have info about vectorized access in runtime functions, e.g. blockIterGroupedYdimReduce()`

Reply:
I understand that. Does that matter?

Reply:
Yes, that matters. If a tv is consumbed by a `IterGroupedReduction`, then it is a input to runtime function `blockIterGroupedYdimReduce` and will be vectorized loaded to smem within that runtime function, so we need to allocate it as aligned array.

Reply:
Yes, I understand that. Does that matter for my question for this?

> It seems strange to me that the kernel summary only has info about global memory tensors but not shared memory. Is there any reason for that?

Reply:
oh, I mean this statement is not true. Kernel summary has info about vectorized r/w for both global and shared memories, if these vectorized r/w were generated by the scheduler.
It does't have the info about vectorized r/w implemented in the runtime functions.

Reply:
I know that. I'm asking why.

Reply:
Becase it doesn't check the runtime functions. These runtime functions are just `strings` when generating kernel.

Reply:
It isn't necessary to do so, right? Your PR doesn't do that either. You check some condition in CudaCodeGen. All I'm asking is why it cannot be done when generating the kernel summary. Let me say this again:

> In general, Kernel should already have the correct vectorization factor and CudaCodeGen should just be a straightforward printer.

Reply:
So you are suggesting doing option-2?

> Another option, is adding these tvs with vectorized shared memory access to kernel_->summary().vectorized_accesses.

No problem. we can do that.